presto-main/src/main/java/com/facebook/presto/execution/scheduler/NodeMap.java

```diff
@@ -84,4 +91,16 @@ public NodeMap(
     {
         return allNodesByHostAndPort;
     }
+
+    public NodeProvider getActiveNodeProvider(NodeSelectionHashStrategy nodeSelectionHashStrategy)
+    {
+        switch (nodeSelectionHashStrategy) {
+            case MODULAR_HASHING:
+                return new ModularHashingNodeProvider(activeNodes);
```

Comment:
Why don't we just use `allNodes` to keep the original logic? 
then we can also remove the hacky part 166~168 in SimpleNodeSelector.java

Reply:
The original logic uses `activeNodes` for hard affinity, and `allNodes` only with soft affinity. `ActiveNodes` should be the right one to use. `allNodes` is the hack.